src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -0,0 +1,156 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="AddDependenciesToProjectJson" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+  <UsingTask TaskName="EvaluateMSBuildProject" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+  <UsingTask TaskName="GatherAssemblyVersionsFromPackageDrop" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+
+  <ItemGroup>
+    <_PackagesDrops Include="$(PackagesDrops)" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <!-- We have to add BeforeResolveReferences to CleanDependsOn otherwise CleanDependsOn will Call AssignProjectConfigurations before we get a chance
+         to swap out the ProjectReferences if we are swapping project references to package dependencies -->
+    <CleanDependsOn>
+      BeforeResolveReferences;
+      $(CleanDependsOn)
+    </CleanDependsOn>
+    <ResolveAssemblyReferencesDependsOn>
+      $(ResolveAssemblyReferencesDependsOn);
+      DetermineProjectJsonPath;
+      ResolveNuGetPackages;
+    </ResolveAssemblyReferencesDependsOn>
+  </PropertyGroup>
+
+  <Target Name="GenerateTestProjectJson" DependsOnTargets="ExtractProjectReferences;GatherAssemblyVersionsFromPackagesDrops;GatherProjectReferenceForProjectJson;DetermineProjectJsonPath;AddDependenciesToProjectJson" />
+
+  <!-- Override targets for building against packages -->
+  <Target Name="BuildAndTest" DependsOnTargets="GenerateTestProjectJson;Build;Test" />
+  <Target Name="RebuildAndTest" DependsOnTargets="GenerateTestProjectJson;Rebuild;Test" />
+
+  <Target Name="DetermineProjectJsonPath">
+    <PropertyGroup>
+      <OriginalProjectJson>$(ProjectJson)</OriginalProjectJson>
+      <!-- We can't put the generated project.json in the output path or NuGet will interpret Microsoft.Win32.Primitives as a "project"
+           type instead of a "package" type when generating the project.lock.json file, so we have to place our project.json in a different path location. -->
+      <ProjectJson>$(GeneratedProjectJsonDir)\$(MSBuildProjectName)\json\project.json</ProjectJson>
+      <GeneratedProjectLockJson>$(GeneratedProjectJsonDir)\$(MSBuildProjectName)\json\project.lock.json</GeneratedProjectLockJson>
+      <!-- Not all projects generate a new project.json, so only update the project.lock.json if it has been restored. -->
+      <ProjectLockJson Condition="Exists('$(GeneratedProjectLockJson)')">$(GeneratedProjectLockJson)</ProjectLockJson>
+    </PropertyGroup>
+    <Message Condition="Exists('$(ProjectLockJson)')" Text="Using '$(ProjectLockJson)'" />
```

Comment:
Probably should just remove this or make it low importance and remove the condition.